dataflow/gemma/README.md

```diff
@@ -0,0 +1,150 @@
+
+The Keras implementation of the Gemma models has a `generate` method
+that generates text based on a prompt. To route the prompts correctly, use this function in the `run_inference` function.
+
+```
+class GemmaModelHandler(ModelHandler[str,
+                                     PredictionResult,GemmaCausalLM
+                                     ]):
+    def __init__(
+        self,
+        model_name: str = "",
+    ):
+        """ Implementation of the ModelHandler interface for spaCy using text as input.
+
+        Example Usage::
+
+          pcoll | RunInference(GemmaModelHandler())
+
+        Args:
+          model_name: The Gemma model uri.
+        """
+        self._model_name = model_name
+        self._env_vars = {}
+    def share_model_across_processes(self)  -> bool:
+        return True
+
+    def load_model(self) -> GemmaCausalLM:
+        """Loads and initializes a model for processing."""
+        return keras_nlp.models.GemmaCausalLM.from_preset(self._model_name)
+
+    def run_inference(
+        self,
+        batch: Sequence[str],
+        model: GemmaCausalLM,
+        inference_args: Optional[Dict[str, Any]] = None
+    ) -> Iterable[PredictionResult]:
+        """Runs inferences on a batch of text strings.
+
+        Args:
+          batch: A sequence of examples as text strings.
+          model:
+          inference_args: Any additional arguments for an inference.
+
+        Returns:
+          An Iterable of type PredictionResult.
+        """
+        # Loop each text string, and use a tuple to store the inference results.
+        predictions = []
+        for one_text in batch:
+            result = model.generate(one_text, max_length=64)
+            predictions.append(result)
+        return [PredictionResult(x, y) for x, y in zip(batch, predictions)]
+```
+
+### Formatting DoFn
+
+The output from a keyed model handler is a tuple of the form `(key, PredictionResult)`. To format that output into a string before sending it to the answer Pub/Sub topic, use an extra `DoFn`.
+
+```
+class FormatOutput(beam.DoFn):
+  def process(self, element, *args, **kwargs):
+    yield "Key : {key}, Input: {input}, Output: {output}".format(key=element[0], input=element[1].example, output=element[1].inference)
+```
+
+## Start the pipeline
+Run the following code from the directory to start the Dataflow streaming pipeline. Replace `$PROJECT`, `$GCS_BUCKET`, `$REGION`, `$CONTAINER_URI`, `$INPUT_TOPIC`, and `$OUTPUT_TOPIC` with the Google Cloud Project resources you created earlier. It may take around 5 minutes for the worker to start up and begin accepting messages from the input Pub/Sub topic. 
+
+```
+python custom_mode_gemma.py \
+--runner=dataflowrunner \
+--project=$PROJECT \
+--temp_location=$GCS_BUCKET \
+--region=$REGION \
+--machine_type="g2-standard-4 \
+--sdk_container_image=$CONTAINER_URI \ 
+--disk_size_gb=200 \
+--dataflow_service_options="worker_accelerator=type:nvidia-l4;count:1;install-nvidia-driver" \
```

Comment:
do l4 have freebie quota? if not, cst might need to get quota in the region of their choice.

Reply:
Added a call-out to check l4 quota in their project